src/utils/wrapper.ts

```diff
@@ -0,0 +1,68 @@
+import * as grpc from 'grpc';
+import * as apm from 'elastic-apm-node';
+import * as _ from 'lodash';
+import { Severity, log } from "./logger";
+import { statusToString, validateGrpcError } from './errors/grpc.status';
+import { ApplicationError } from './errors/errors';
+
+/**
+ * extracts the wanted information for the logger from the response.
+ * for example: extracts only the file ids from the returned array of files.
+ * @param res - the result of the method called in file.grpc
+ */
+function extractResLog(res: any): object {
+    if (!res) return {};
+
+    return _.cloneDeep(res);
+}
+
+/**
+ * extracts the wanted information for the logger from the request.
+ * for example: extracts only relevant fields in the queryFile.
+ * @param req - the call.request received in the service.
+ */
+function extractReqLog(req: any): object {
+    return _.cloneDeep(req);
+}
+
+
+export function getCurrTraceId(): string {
+    try {
+        return apm.currentTransaction ? apm.currentTransaction.traceparent.split('-')[1] : '';
+    } catch (err) {
+        // Should never get here. The log is set after apm starts.
+        log(Severity.ERROR, 'error on getting trace id', 'apm');
```

Comment:
Moreover, notice that logging this error as `error on getting trace id`  gives you no information other than that error just accrued. It probably won't help you understand why or during what it has happened.